### ISSUE TITLE
Elassandra 6.2.3.7 does not work

### ISSUE THREAD

**Issue by eran1234** [strapdata/elassandra#243]
Elasticsearch version: 6.2.3.7 

Plugins installed: [ ]

JVM version (java -version):1.8

OS version (uname -a if on a Unix-like system): debian 9

I tried to install and run Elassandra 6.2.3.7 using the tarball installation instructions on a new X64 Debian 9 and it did not work.  I am using Java 1.8 and I set the JAVA_HOME and CASSANDRA_HOME correctly. 
I tried also the same using version 6.2.3.6 but it failed. 

Elassandra 5.x was installed and run without any issue using the tarball instructions.
Installing Elassandra 6.2.3.7 using deb file worked fine.

I used strace to run Elassandra 6.2.3.7 when installing using tarball instructions and I see it exits with code 1 without any explanation to why.

What am I missing? Was installation of Elassandra 6.2.3.7 was tested using tarball instructions?

Thanks,
Eran

**Comment by DBarthe** [comment]
Hello,

Can you provide the content cassandra logs after running the tarball ?
(`$CASSANDRA_HOME/logs/system.log`) ?

Did you start elassandra 6.2.3.7 on a fresh cluster, I mean with no
existing data ? Or did you upgrade from a previous version ?

**Comment by eran1234** [comment]
Hi,

There was no system.log file created.
I had a new debian 9 and nothing on it. I did not had Elassandra before on that device. 

Once I failed I tried the version 6.2.3.6 which failed too. Since I am using Elassandra 5.x on other device, I tried to install it on the new device and it worked fine.

Only the deb package worked of version 6.2.3.7.

**Comment by DBarthe** [comment]
Probably the script `bin/cassandra` failed before it starts the java process.

For troubleshooting, you could add the instruction `set -xe` on top of the `bin/cassandra` and start again.
It will print every steps of the script execution.

**Comment by eran1234** [comment]
[elassandra_6_2_3_7_with_set_xe.txt](https://github.com/strapdata/elassandra/files/2626162/elassandra_6_2_3_7_with_set_xe.txt)

Please find it attached.

**Comment by vroyer** [comment]
Your log is incomplet or for some reasons the start script `bin/cassandra` ends before launching the JVM. Please check your environment.

**Comment by eran1234** [comment]
What do you mean by check the environment? 
I have java_home and cassandra_home set and it works fine on elassandra 5 but when I switch them to work with elassandra 6 it does not work.

Am I missing anything?

**Comment by dbapramod882** [comment]
Not sure, if i am correct.

I see, that from the log you are using java version `openjdk version "1.8.0_181"`

Can you try using lower version of java like `java version "1.8.0_66"`

**Comment by eran1234** [comment]
It did not work. See attachment. 
Please note that the latest Elassandra 5 works fine just 6 has issues therefore I do not believe it has something to do with environment.

Were you able to install Elassandra 6 using the tar file on a new device?
[elassandra_6_2_3_7_with_set_xe_jvm_oracle_1_8_1_u45.txt](https://github.com/strapdata/elassandra/files/2630393/elassandra_6_2_3_7_with_set_xe_jvm_oracle_1_8_1_u45.txt)

**Comment by eran1234** [comment]
Have you managed to duplicate the installation issue of elassandra 6.2.3.7?

You can create a virtual machine x64, install debian9 using amd64 installation file and after that to try to install elassandra 6.2.3.7

Is there anything I can provide you to help you find the root cause?

**Comment by blnusr** [comment]
Hi,

I can also confirm what eran1234 submitted. On the same machine, elassandra-5.5.0.24 is working and elassandra-6.2.3.7 isn't. I have checked if the CASSANDRA_HOME/bin/cassandra Shell-Scripts are identical between those two versions, and they are binary identical.  So the same environment is good for elassandra-5.x but it isn't working for elassandra-6.x.
I have also installed a new clean VM with debian9, installed OpenJDK8, set JAVA_HOME and CASSANDRA_HOME, then started elassandra-5.5.0.24 and got the TCP port 9200 and was able to get a response via 'curl -X GET "http://127.0.0.1:9200"'.
But with elassandra-6.2.3.7 no logfiles have been created and no process-ID has been created and of course no port available.
I have checked that CASSANDRA_HOME is always pointing to the correct installation directory by using the command env. I repeated the tests with few different Oracle-JVMs and OpenJDK8. I have also experienced the same issue on aarch64 architecture.
Are maybe some missing privileges the root cause for not starting the service?
